### PR TITLE
Update coverage/tree route to parse flags and components filters 

### PR DESCRIPTION
**Pull request by rohitvinnakota-codecov**
API changes to propagate totals data to the sunburst chart on the front end. 

The chart calls the `/coverage/tree endpoint`, which does not currently filter on flags or components. This change makes it possible for it to do so. We do something similar in [commit.py](https://github.com/codecov/codecov-api/blob/main/graphql_api/types/commit/commit.py).

There will be a follow up PR on gazebo that passes flags and components as query params. 

Example call on staging 
Screenshot 2024-04-15 at 2 46 20 PM

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Review by barnett-yuxiang — COMMENTED** [review]
LGTM

File: api/internal/tests/views/test_coverage_viewset.py [hunk 1]
```diff
@@ -286,3 +296,39 @@ def test_tree_missing_report(self, build_report_from_commit):
...
+                {
+                    "component_id": "c1",
+                    "name": "ComponentOne",
+                    "paths": ["dne.py"],
+                }
+            ),
+        ]
+        build_report_from_commit.return_value = sample_report()
+        res = self._tree(components="Does_not_exist")
+        assert res.status_code == 404
```
Comment (ajay-sentry) [inline api/internal/tests/views/test_coverage_viewset.py:334]: Should we also add 1 more test for the flag filtering?

**Review by ajay-sentry — APPROVED** [review]
test comment, logic looks good

**Review by barnett-yuxiang — APPROVED** [review]
good

**Pull request merged**